src/contacts/EditContact/EditContactContainer.js

```diff
@@ -21,11 +21,15 @@ class EditContactContainer extends Component {
   onClose = (contactId = this.props.match.params.id) => {
     const { orgId, mutator } = this.props;
     const starting = orgId ? `/organizations/${orgId}` : '/organizations';
+    const query = contactId
```

Comment:
You ignore create organization flow here. As I see you can use approach with generating `starting` and `ending` of required path

Reply:
also the difference in logic will be in just `contacts` or `interface` part, so it could be leveraged into an util function of `getBackPath(orgId, itemId, itemPath)`